Site Settings API: Expose `show_context` and `show_date` options

mashikag opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/wp-calypso/issues/71593

It is necessary to expose those options to implement the **Related posts** setting in the new **Reading Settings** page.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Extend site settings API to expose `show_context` and `show_date` related posts options for both reading (`GET /sites/%s/settings`) and writing (`POST /sites/%s/settings`)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Apply the PR to your sandbox
* Sandbox `public-api.wordpress.com`
* Go to wpcom API developer console

#### Testing `POST /sites/%s/settings/`
 * Select `WP.COM API` - `v1.1` - `POST` - `/sites/$site/settings` where `$site` can be your test site's domain or site ID
 * Set `show_context` and/or `show_date` body args to either `true` or `false`
 * Execute the call
 * Validate the response
 * You may double check the values were persisted by executing the `GET /sites/%s/settings/`. See below for instructions.

#### Testing `GET /sites/%s/settings/`
 * Select `WP.COM API` - `v1.1` - `GET` - `/sites/$site/settings` where `$site` can be your test site's domain or site ID
 * Execute the call
 * Validate the response. Ensure the `jetpack_relatedposts_show_context` and the `jetpack_relatedposts_show_date` are returned with the correct values

mashikag committed:
[not verified] Site Settings API: Expose related posts

mashikag committed:
Site Settings API: Expose related posts

mashikag committed:
Site Settings API: Expose related posts  option

mashikag committed:
[not verified] Site Settings API: Expose related posts  option

mashikag committed:
changelog

mashikag committed:
[not verified] changelog

adimoldovan committed:
E2E tests: remove broken selection of complete plan (#28232)

* Remove selection of complete plan in connection flow as it's broken and not used anywhere anyway.

mashikag committed:
[not verified] Fix return value on related posts option update

mashikag committed:
Fix return value on related posts option update

mashikag committed:
[not verified] Handle undefined  or  options

mashikag committed:
Handle undefined  or  options

mashikag committed:
[not verified] Remove unnecassary bool cast

mashikag committed:
Remove unnecassary bool cast

ivan-ottinger commented on projects/plugins/jetpack/changelog/update-expose-related-posts-show-date-option:
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: other
+
+Site Settings API: Expose jetpack's related posts 'show_date' and 'show_context' options
```
Small typo we can fix:

```suggestion
Site Settings API: Expose Jetpack's related posts 'show_date' and 'show_context' options
```

renatoagds committed:
VideoPress: Allow edit privacy on video details page (#28240)

* VideoPress: Add selector for privacy in edit page

* changelog

ivan-ottinger reviewed (commented):
Nice! Works as expected. 🙂👍🏼 

Details:
- Tested on Simple, Jetpack and Atomic (both `GET` and `POST` for each)
- Shared one tiny suggestion in a comment below.
- Did not notice any PHP errors / warnings.
- We noticed that when we change e.g. `show_context` to `false`, it will switch `show_headline` and `show_thumbnails` to `false` as well. This is currently expected behavior, although probably better addressed in a different PR - as we discussed in p1673355980792549/1673349549.496449-slack-C02TCEHP3HA.

mashikag committed:
[not verified] Update projects/plugins/jetpack/changelog/update-expose-related-posts-show-date-option

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

mashikag committed:
Update projects/plugins/jetpack/changelog/update-expose-related-posts-show-date-option

Co-authored-by: Ivan Ottinger <25105483+ivan-ottinger@users.noreply.github.com>

jeherve commented on projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php:
```diff
@@ -365,6 +367,8 @@ public function get_settings_response() {
 						'jetpack_sync_non_public_post_stati' => (bool) Jetpack_Options::get_option( 'sync_non_public_post_stati' ),
 						'jetpack_relatedposts_allowed'     => (bool) $this->jetpack_relatedposts_supported(),
 						'jetpack_relatedposts_enabled'     => (bool) $jetpack_relatedposts_options['enabled'],
+						'jetpack_relatedposts_show_context' => isset( $jetpack_relatedposts_options['show_context'] ) && $jetpack_relatedposts_options['show_context'],
+						'jetpack_relatedposts_show_date'   => isset( $jetpack_relatedposts_options['show_date'] ) && $jetpack_relatedposts_options['show_date'],
```
Should we cast those 2 as booleans like we do above and below, for consistency?

mashikag replied:
Sure. I suppose it is there only for human readability, right? I actually had the casting and removed it because I thought there was no need for it. 😅

ivan-ottinger replied:
Btw, I tested the PR with and also without casting - as I started testing it when the `(bool)`s were still there. 🙂

mashikag replied:
BTW the casting done on the lines below seems incorrect to me too. It only casts the result of `isset( $jetpack_relatedposts_options['show_headline'] )`. The actual value that is being returned ends up being `"1"`, `"0"` or `false`.

jeherve replied:
> BTW the casting done on the lines below seems incorrect to me too

That's a good point actually, nice catch! We could update all those to be correct, and to use your format? We wouldn't even need to cast as booleans then, as you pointed out. I was mostly concerned about consistency, with your original approach the (bool) isn't really necessary.

Could you update all those in this PR, or in a follow-up PR?

anomiex replied:
For that matter, `isset( $foo ) && (bool) $foo` is the same as `! empty( $foo )` if you want to simplify the code a bit.

mashikag replied:
@jeherve let me do it with this one. 🙂 

@anomiex good point. I will do as per your suggestion. 👍

retrofox committed:
VideoPress: set Video URL in replace control based on privacy (#28239)

* [not verified] set Video URL based on privacy

* changelog

* fix private URL

mashikag committed:
Bring back the bool cast

mashikag committed:
[not verified] Bring back the bool cast

adimoldovan committed:
E2E tests: fix waitForResponse predicates (#28262)

retrofox committed:
VideoPress: minor TS enhancements in the useSearchParams() hook (#28250)

* set SearchParamNameProp type

* minor jsdoc changes

* changelog

matticbot committed:
Update dependency @storybook/testing-react to v1.3.0 (#28269)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency qrcode.react to v3 (#28270)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency react-page-visibility to v7 (#28272)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

coder-karen committed:
Calendar library and Tonesque: various cleanup (#28274)

* [not verified] Calendar library: various cleanup

* [not verified] Correct logic

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

* [not verified] Update projects/plugins/jetpack/modules/shortcodes/upcoming-events.php

Co-authored-by: Karen Attfield <karenlattfield@gmail.com>

Co-authored-by: Brandon Kraft <public@brandonkraft.com>

retrofox committed:
VideoPress: clean video attributes that are not options when replacing the video file (#28249)

* clean attrs that are not options when replacing

* changelog

* fix setting defauls attribute values issue

* replace block when replacing with a new file

coder-karen committed:
Backporting Jetpack, Social and Backup readmes and changelog entries (#28271)

* [not verified] Updating changelog for 11.7 release

* [not verified] Updating readme.txt for 11.7 release

* [not verified] Updating changelog and readme for 1.4.3

* [not verified] Cleaning up changelog and readme for 1.6.0

* Updating stable tag in readme

* Updating changelog to link to release post

* Updating Social and Backup readme version

* Cleaning up Social and Backup readme and changelog

matticbot committed:
Update storybook monorepo to v6.5.15 (#28268)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

matticbot committed:
Update dependency postcss to v8.4.21 (#28278)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

jeherve committed:
Social Menu / Social Icons Widget: add Mastodon (#28175)

thingalon committed:
[Boost] Don't show <1x on images that are exactly the right size (#28245)

* Mark exact-sized images with a check mark

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

matticbot committed:
Update dependency storybook-addon-mock to v3 (#28279)

* Update dependency storybook-addon-mock to v3
* Update use of storybook-addon-mock for changes in v3

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

thingalon committed:
Remove the Image Guide feature-flag (#28244)

* Enable the Image Guide without a feature-flag

* changelog

Co-authored-by: Mark George <thingalon@gmail.com>

mashikag committed:
Fix up some options assignments

mashikag committed:
[not verified] Fix up some options assignments

jeherve committed:
Connection: add new method to fetch a connected site's blog ID. (#28208)

danielbachhuber committed:
Form Block: Add a few keywords to the 'Newsletter Sign-up' variation (#28204)

jeherve committed:
Blocks: Add AI-powered blocks for WordPress.com customers (#28043)

Co-authored-by: artpi <artur.piszek@gmail.com>

coder-karen committed:
Revert "Tiled Gallery block: Adding styles to squareish tiled gallery images on WoA sites (#27858)" (#28287)

This reverts commit 5e3d04a31e036cbcb2b2abcd52bafaf1127d7ccf.

retrofox committed:
VideoPress: fix render player once file uploads issue (#28296)

* update timeour reference when cleaning the timer

* changelog

adimoldovan committed:
E2E tests: new test for subscribe block (#28289)

nateweller committed:
Protect: Poll for scan status when scanner is idle and no lastChecked value is available (#28275)

samiff committed:
Jetpack: 11.8-a.1 release (#28298)

* Jetpack: 11.8-a.1 release changelog

* Jetpack: init 11.8-a.2

* Jetpack: refresh prerelease branch before backport

nateweller committed:
Protect: Round ProgressBar percentage values (#28273)

anomiex committed:
build: Use PR commit SHA in plugin artifact version numbers (#28294)

We intend for the version number in the plugin artifacts to include the
short hash for the PR head, but the logic used was picking up the merge
commit GitHub creates instead. Fix that.

samiff committed:
Jetpack: 11.8-a.1 changelog editorial (#28300)

* Jetpack: 11.8-a.1 changelog editorial

* changelog

* Fix missing backtick messing with MD display.

adimoldovan committed:
E2E tests: remove check for post page (#28299)

jeherve committed:
Blaze: only display UI to English customers. (#28266)

renatoagds committed:
VideoPress: Fix app breaking when deleting last page video. (#28281)

* VideoPress: Push back page if deleting last video

* changelog

* VideoPress: Update totalPages and total, and move to action

matticbot committed:
Update react-router monorepo (#28303)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

mashikag committed:
Merge branch 'update/expose-related-posts-show-date-option' of github.com:Automattic/jetpack into update/expose-related-posts-show-date-option

mashikag commented:
OMG 🤦‍♂️ 

@ivan-ottinger and @jeherve I am so sorry for this mayhem I have caused with my late night (trying to make myself feel better) latest trunk rebase.

This PR branch just became too big of a mess to be fixing it here. Instead I have just created a new branch out of my last [commit](https://github.com/Automattic/jetpack/pull/28251/commits/3ca570436dcda524c9598a9e708e6f8c7e1ab866) I did before the fatal rebase. 

The new PR: https://github.com/Automattic/jetpack/pull/28305

ivan-ottinger commented:
No worries, Maciej! I am going to take a look at the new PR. :)
